插件冲突，建议使用共通上游

与使用：[nonebot-plugin-tortoise-orm](https://github.com/kexue-z/nonebot-plugin-tortoise-orm) 插件有冲突，覆盖了原先的数据库链接

改进建议：立即使用 [nonebot-plugin-tortoise-orm ](https://github.com/kexue-z/nonebot-plugin-tortoise-orm)

相关链接：https://github.com/kexue-z/nonebot-plugin-setu-now/issues/54#issuecomment-1685781949



<details><summary>错误日志：</summary>
<p>


```
PS D:\临时\bot\setu-test> nb run
使用 Python: D:\临时\bot\setu-test\.venv\Scripts\python.exe
08-21 15:13:35 [SUCCESS] nonebot | NoneBot is initializing...
08-21 15:13:35 [INFO] nonebot | Current Env: prod
08-21 15:13:35 [SUCCESS] nonebot | Succeeded to load plugin "nonebot_plugin_localstore"
08-21 15:13:35 [WARNING] nonebot_plugin_tortoise_orm | 没有设置数据库地址, 使用 sqlite:///C:\Users\****\AppData\Local\nonebot2\db.sqlite3
08-21 15:13:35 [SUCCESS] nonebot | Succeeded to load plugin "nonebot_plugin_tortoise_orm"
08-21 15:13:35 [SUCCESS] nonebot_plugin_tortoise_orm | 数据库: 添加模型: nonebot_plugin_setu_now.database
08-21 15:13:35 [SUCCESS] nonebot | Succeeded to load plugin "nonebot_plugin_setu_now"
08-21 15:13:37 [SUCCESS] nonebot | Succeeded to load plugin "nonebot_plugin_apscheduler"
08-21 15:13:37 [SUCCESS] nonebot | Succeeded to load plugin "nonebot_plugin_learning_chat"
08-21 15:13:37 [SUCCESS] nonebot | Running NoneBot...
08-21 15:13:37 [INFO] uvicorn | Started server process [19960]
08-21 15:13:37 [INFO] uvicorn | Waiting for application startup.
08-21 15:13:37 [SUCCESS] nonebot_plugin_tortoise_orm | 数据库: 连接成功 URL: sqlite:///C:\Users\****\AppData\Local\nonebot2\db.sqlite3
08-21 15:13:37 [INFO] nonebot_plugin_learning_chat | [群聊学习]数据库连接<g>成功</g>
08-21 15:13:37 [INFO] nonebot_plugin_apscheduler | Scheduler Started
08-21 15:13:37 [INFO] uvicorn | Application startup complete.
08-21 15:13:37 [INFO] uvicorn | Uvicorn running on http://127.0.0.1:8080 (Press CTRL+C to quit)
08-21 15:13:38 [INFO] uvicorn | ('127.0.0.1', 58662) - "WebSocket /onebot/v11/ws" [accepted]
08-21 15:13:38 [INFO] nonebot | OneBot V11 | Bot 1234connected
08-21 15:13:38 [INFO] websockets | connection open
08-21 15:14:11 [SUCCESS] nonebot | OneBot V11 1234| [message.group.normal]: Message 10004 from 1234@[群:1234] '色图'
08-21 15:14:11 [INFO] nonebot | Event will be handled by Matcher(type='message', module=nonebot_plugin_setu_now) 
08-21 15:14:11 [INFO] nonebot | Matcher(type='message', module=nonebot_plugin_setu_now) running complete
08-21 15:14:11 [ERROR] nonebot | Running Matcher(type='message', module=nonebot_plugin_setu_now) failed.
Traceback (most recent call last):
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\tortoise\backends\sqlite\client.py", line 34, in translate_exceptions_
    return await func(self, query, *args)
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\tortoise\backends\sqlite\client.py", line 141, in execute_query
    rows = await connection.execute_fetchall(query, values)
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\aiosqlite\core.py", line 203, in execute_fetchall
    return await self._execute(self._execute_fetchall, sql, parameters)
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\aiosqlite\core.py", line 129, in _execute
    return await future
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\aiosqlite\core.py", line 102, in run
    result = function()
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\aiosqlite\core.py", line 81, in _execute_fetchall
    cursor = self._conn.execute(sql, parameters)
sqlite3.OperationalError: no such table: white_list

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 15, in <module>
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\nonebot\__init__.py", line 333, in run
    get_driver().run(*args, **kwargs)
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\nonebot\drivers\fastapi.py", line 199, in run
    uvicorn.run(
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\uvicorn\main.py", line 587, in run
    server.run()
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "C:\Users\****\AppData\Local\miniconda3\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\****\AppData\Local\miniconda3\lib\asyncio\base_events.py", line 636, in run_until_complete   
    self.run_forever()
  File "C:\Users\****\AppData\Local\miniconda3\lib\asyncio\windows_events.py", line 321, in run_forever       
    super().run_forever()
  File "C:\Users\****\AppData\Local\miniconda3\lib\asyncio\base_events.py", line 603, in run_forever
    self._run_once()
  File "C:\Users\****\AppData\Local\miniconda3\lib\asyncio\base_events.py", line 1906, in _run_once
    handle._run()
  File "C:\Users\****\AppData\Local\miniconda3\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\nonebot\message.py", line 467, in check_and_run_matcher    
    await _run_matcher(
> File "D:\临时\bot\setu-test\.venv\lib\site-packages\nonebot\message.py", line 419, in _run_matcher
    await matcher.run(bot, event, state, stack, dependency_cache)
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\nonebot\internal\matcher\matcher.py", line 759, in run     
    await self.simple_run(bot, event, state, stack, dependency_cache)
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\nonebot\internal\matcher\matcher.py", line 734, in simple_run
    await handler(
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\nonebot\dependencies\__init__.py", line 104, in __call__   
    values = await self.solve(**kwargs)
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\nonebot\dependencies\__init__.py", line 220, in solve      
    values = await asyncio.gather(
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\nonebot\dependencies\__init__.py", line 209, in _solve_field
    value = await cast(Param, field.field_info)._solve(**params)
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\nonebot\internal\params.py", line 166, in _solve
    return await task
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\nonebot_plugin_setu_now\r18_whitelist.py", line 15, in get_group_white_list_record
    res = await GroupWhiteListRecord.get_or_none(group_id=event.group_id)
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\tortoise\queryset.py", line 1008, in _execute
    instance_list = await self._db.executor_class(
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\tortoise\backends\base\executor.py", line 131, in execute_select
    _, raw_results = await self.db.execute_query(query.get_sql())
  File "D:\临时\bot\setu-test\.venv\lib\site-packages\tortoise\backends\sqlite\client.py", line 36, in translate_exceptions_
    raise OperationalError(exc)
tortoise.exceptions.OperationalError: no such table: white_list
08-21 15:14:16 [INFO] uvicorn | Shutting down
08-21 15:14:16 [WARNING] nonebot | OneBot V11 | WebSocket for Bot 1234closed by peer
08-21 15:14:16 [INFO] websockets | connection closed
08-21 15:14:16 [INFO] uvicorn | Waiting for application shutdown.
08-21 15:14:16 [SUCCESS] nonebot_plugin_tortoise_orm | 数据库: 断开链接
08-21 15:14:16 [INFO] nonebot_plugin_learning_chat | [群聊学习]数据库断开连接<g>成功</g>
08-21 15:14:16 [INFO] nonebot_plugin_apscheduler | Scheduler Shutdown
08-21 15:14:16 [INFO] uvicorn | Application shutdown complete.
08-21 15:14:16 [INFO] uvicorn | Finished server process [19960]
```

</p>
</details>

--- a/nonebot_plugin_learning_chat/__init__.py
+++ b/nonebot_plugin_learning_chat/__init__.py
@@ -1,9 +1,7 @@
 from nonebot import require
 
 require("nonebot_plugin_tortoise_orm")
-from nonebot_plugin_tortoise_orm import add_model
 
-add_model("nonebot_plugin_learning_chat.models")
 
 import asyncio
 import random

--- a/nonebot_plugin_learning_chat/__init__.py
+++ b/nonebot_plugin_learning_chat/__init__.py
@@ -1,9 +1,22 @@
+from nonebot import require
+
+require("nonebot_plugin_tortoise_orm")
+from nonebot_plugin_tortoise_orm import add_model
+
+add_model("nonebot_plugin_learning_chat.models")
+
 import asyncio
 import random
 import time
 
 from nonebot import on_message, require, logger, get_adapter
-from nonebot.adapters.onebot.v11 import GroupMessageEvent, GROUP, Message, ActionFailed, Adapter
+from nonebot.adapters.onebot.v11 import (
+    GroupMessageEvent,
+    GROUP,
+    Message,
+    ActionFailed,
+    Adapter,
+)
 from nonebot.params import Arg
 from nonebot.plugin import PluginMetadata
 from nonebot.rule import Rule
@@ -23,9 +36,7 @@ __plugin_meta__ = PluginMetadata(
     type="application",
     homepage="https://github.com/CMHopeSunshine/nonebot-plugin-learning-chat",
     supported_adapters={"~onebot.v11"},
-    extra={
-        "author": "惜月"
-    },
+    extra={"author": "惜月"},
 )
 
 

--- a/nonebot_plugin_learning_chat/models.py
+++ b/nonebot_plugin_learning_chat/models.py
@@ -1,3 +1,11 @@
+from nonebot_plugin_tortoise_orm import add_model
+
+add_model(
+    __name__,
+    db_name="learning_chat",
+    db_url="sqlite://data/learning_chat/learning_chat.db",
+)
+
 import functools
 from functools import cached_property
 from pathlib import Path
@@ -121,7 +129,10 @@ class ChatAnswer(Model):
     """消息列表"""
 
     context: fields.ForeignKeyNullableRelation[ChatContext] = fields.ForeignKeyField(
-        "models.ChatContext", related_name="answers", null=True
+        # db_name.models_name
+        "learning_chat.ChatContext",
+        related_name="answers",
+        null=True,
     )
 
     class Meta:
@@ -143,22 +154,3 @@ class ChatBlackList(Model):
     class Meta:
         table = "blacklist"
         indexes = ("keywords",)
-
-
-# @driver.on_startup
-# async def startup():
-#     try:
-#         await Tortoise.init(
-#             db_url=f"sqlite://{DATABASE_PATH}", modules={"models": [__name__]}
-#         )
-#         await Tortoise.generate_schemas()
-#         log_info("群聊学习", "数据库连接<g>成功</g>")
-#     except Exception as e:
-#         log_info("群聊学习", f"数据库连接<r>失败，{e}</r>")
-#         raise e
-
-
-# @driver.on_shutdown
-# async def shutdown():
-#     await Tortoise.close_connections()
-#     log_info("群聊学习", "数据库断开连接<g>成功</g>")

--- a/nonebot_plugin_learning_chat/models.py
+++ b/nonebot_plugin_learning_chat/models.py
@@ -13,9 +13,9 @@ try:
 except ImportError:
     import jieba
     import jieba.analyse as jieba_analyse
-from tortoise import fields, Tortoise
+from tortoise import fields
 from tortoise.models import Model
-from .config import config_manager, driver, log_info
+from .config import config_manager
 
 from tortoise.connection import ConnectionHandler
 
@@ -35,8 +35,8 @@ ConnectionHandler._init = _init
 
 config = config_manager.config
 
-DATABASE_PATH = Path() / "data" / "learning_chat" / "learning_chat.db"
-DATABASE_PATH.parent.mkdir(parents=True, exist_ok=True)
+# DATABASE_PATH = Path() / "data" / "learning_chat" / "learning_chat.db"
+# DATABASE_PATH.parent.mkdir(parents=True, exist_ok=True)
 JSON_DUMPS = functools.partial(json.dumps, ensure_ascii=False)
 jieba.setLogLevel(jieba.logging.INFO)
 jieba.load_userdict(str(Path(__file__).parent / "genshin_word.txt"))  # 加载原神词典
@@ -145,20 +145,20 @@ class ChatBlackList(Model):
         indexes = ("keywords",)
 
 
-@driver.on_startup
-async def startup():
-    try:
-        await Tortoise.init(
-            db_url=f"sqlite://{DATABASE_PATH}", modules={"models": [__name__]}
-        )
-        await Tortoise.generate_schemas()
-        log_info("群聊学习", "数据库连接<g>成功</g>")
-    except Exception as e:
-        log_info("群聊学习", f"数据库连接<r>失败，{e}</r>")
-        raise e
+# @driver.on_startup
+# async def startup():
+#     try:
+#         await Tortoise.init(
+#             db_url=f"sqlite://{DATABASE_PATH}", modules={"models": [__name__]}
+#         )
+#         await Tortoise.generate_schemas()
+#         log_info("群聊学习", "数据库连接<g>成功</g>")
+#     except Exception as e:
+#         log_info("群聊学习", f"数据库连接<r>失败，{e}</r>")
+#         raise e
 
 
-@driver.on_shutdown
-async def shutdown():
-    await Tortoise.close_connections()
-    log_info("群聊学习", "数据库断开连接<g>成功</g>")
+# @driver.on_shutdown
+# async def shutdown():
+#     await Tortoise.close_connections()
+#     log_info("群聊学习", "数据库断开连接<g>成功</g>")

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -13,7 +13,7 @@ python = "^3.8"
 nonebot2 = { version = "^2.0.0", extras = ["fastapi"] }
 nonebot-adapter-onebot = "^2.1"
 nonebot-plugin-apscheduler = "^0.2.0"
-tortoise-orm = "^0.19.2"
+tortoise-orm = ">=0.20.0"
 jieba = "^0.42.1"
 "ruamel.yaml" = "^0.17.21"
 amis-python = "^1.0.6"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -10,7 +10,7 @@ license = "AGPL"
 
 [tool.poetry.dependencies]
 python = "^3.8"
-nonebot2 = {version = "^2.0.0", extras = ["fastapi"]}
+nonebot2 = { version = "^2.0.0", extras = ["fastapi"] }
 nonebot-adapter-onebot = "^2.1"
 nonebot-plugin-apscheduler = "^0.2.0"
 tortoise-orm = "^0.19.2"
@@ -18,6 +18,7 @@ jieba = "^0.42.1"
 "ruamel.yaml" = "^0.17.21"
 amis-python = "^1.0.6"
 python-jose = "^3.3.0"
+nonebot-plugin-tortoise-orm = ">=0.1.0"
 
 [tool.poetry.dev-dependencies]
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -18,7 +18,7 @@ jieba = "^0.42.1"
 "ruamel.yaml" = "^0.17.21"
 amis-python = "^1.0.6"
 python-jose = "^3.3.0"
-nonebot-plugin-tortoise-orm = ">=0.1.0"
+nonebot-plugin-tortoise-orm = ">=0.1.1"
 
 [tool.poetry.dev-dependencies]
 

